sycl/include/sycl/types.hpp

```diff
@@ -33,14 +26,7 @@
 #endif
 #endif // __clang__
 
-#ifdef __SYCL_USE_EXT_VECTOR_TYPE__
-#error "Undefine __SYCL_USE_EXT_VECTOR_TYPE__ macro"
-#endif
-#ifdef __HAS_EXT_VECTOR_TYPE__
-#if defined(__SYCL_DEVICE_ONLY__) || !defined(__NO_EXT_VECTOR_TYPE_ON_HOST__)
-#define __SYCL_USE_EXT_VECTOR_TYPE__
-#endif
-#elif defined(__SYCL_DEVICE_ONLY__)
+#if !defined(__HAS_EXT_VECTOR_TYPE__) && defined(__SYCL_DEVICE_ONLY__)
```

Comment:
Could you please clarify why we need to check for `__HAS_EXT_VECTOR_TYPE__`? What unique cases it covers?

Reply:
I think I'm misunderstanding your question. 

This full block is
```
#if !defined(__HAS_EXT_VECTOR_TYPE__) && defined(__SYCL_DEVICE_ONLY__)
// This is a soft error. We expect the device compiler to have ext_vector_type
// support, but that should not be a hard requirement.
#error "SYCL device compiler is built without ext_vector_type support"
#endif // __HAS_EXT_VECTOR_TYPE__
```

And we are erroring if the device compiler does not have ext_vector_type support, because we need it. We don't check at all on the host compiler where we are no longer using it.  

Let me know if I'm missing your intent.

Reply:
But we also have it checked a couple of times in the file. And depending on that either use `ext_vector_type` or not.

Reply:
I think I see what you are after.  Yes, now that we are always using using `ext_vector_type` on device and never using it on host, the inclusion of `__HAS_EXT_VECTOR_TYPE` in some of the guards is now redundant. I have removed them. Pretty much just used for that error output now.